### ISSUE TITLE
Consuming addon definitions and module resolution?

### ISSUE THREAD

**Issue by vcipriani** [typed-ember/ember-cli-typescript#56]
First, thank you for this!  I converted my first addon last night and can't wait to work it into more of our code base.  

I'm wondering about the recommended way(s) to have module resolution work in a parent app/addon when it is consuming an addon written in typescript.  For example, addon-b has addon-a as a dependency and tries to `import thing from addon-a/utils/foo`.  I'm currently adding the following to addon-b's tsconfig.  This appears to work fine, but it requires explicit code for every addon consumed.  Is there a better way?

Thanks and I'd be happy to submit a PR for a doc update w/ whatever approach is recommended.

```json
paths: {
   "addon-a/*": [
        "node_modules/addon-a/addon/*"
   ]
}
```

**Comment by chriskrycho** [comment]
My plan here is actually to have addons do a build step which exports their typings as standalone definition files. The reason is that otherwise people will be downloading multiple copies of the TS compiler just for addons and it weighs a *lot*.

This is a really important issue for that, though, because I hadn't yet considered what we'll need to do to make sure the resolution works for those kinds of automatically generated exports. And I confess: I'm profoundly ignorant of the various kinds of transformations you can do in terms of import paths. I'm hoping @rwjblue will have some helpful input there.

In the meantime, I don't have any idea how to make this easier other than to use `tsc` with the flag for emitting typings and exporting those in a manually configured npm build step along the same lines we're planning to make automatic in the future, but I'll think on it.

**Comment by chriskrycho** [comment]
Resolved in #93.
